src/hooks/context/contextService.ts

```diff
@@ -0,0 +1,46 @@
+import { injectable } from 'tsyringe';
+import { Context, ContextFileNotFoundError,KeyNotFoundError } from './models';
+import { CONTEXTFILE_PATH } from '../../constants';
+import * as fs from 'fs';
+import * as path from 'path';
+import { SpecificationFile } from '../validation';
+
+@injectable()
+export class ContextService {
+	loadContextFile(): Context {
+		if (!fs.existsSync(CONTEXTFILE_PATH)) throw new ContextFileNotFoundError();
+		return JSON.parse(fs.readFileSync(CONTEXTFILE_PATH, 'utf-8')) as Context;
+	}
+
+	deleteContextFile() {
+		if(fs.existsSync(CONTEXTFILE_PATH)) fs.unlinkSync(CONTEXTFILE_PATH);
+	}
+
+	save(context: Context) {
+		fs.writeFileSync(CONTEXTFILE_PATH, JSON.stringify(context), { encoding: 'utf-8' });
```

Comment:
we should also validate if file exists when saving the context, otherwise it doesn't make sense to save it

Reply:
This I had a long thought when I started with this, but then I came to the conclusion that context will only persist path, and we have other features to check if the file exists or not because validate hook is basically doing that.

Reply:
yeah, but the problem is that if I made a mistake in path when adding it, I learn about it when validating, so too late, I should learn about it the moment I create it

Reply:
Yeah, so I should have a check that if that file exists.